### PR TITLE
Support send_params option

### PR DESCRIPTION
**Pull request by tombruijn**
Don't store params on the transaction when the config option is set to
`false`. Ensure feature parity with the Ruby gem.

### CODE REVIEW

File: lib/appsignal/transaction.ex [hunk 1]
```diff
@@ -253,7 +253,19 @@ defmodule Appsignal.Transaction do
   @spec set_sample_data(Transaction.t() | nil, String.t(), any) :: Transaction.t()
   def set_sample_data(nil, _key, _payload), do: nil
 
+  def set_sample_data(%Transaction{} = transaction, "params" = key, payload) do
+    if config()[:send_params] do
+      do_set_sample_data(transaction, key, payload)
```
Comment (jvanbaarsen) [inline lib/appsignal/transaction.ex]: What would be the reason we have the `do_set` and the `set` method? Wouldn't only tghe `set_sample_data` be enough?

Comment (tombruijn) [reply lib/appsignal/transaction.ex]: calling `set_sample_data` again would cause it to get trapped in an infinite loop, right?

Comment (jvanbaarsen) [reply lib/appsignal/transaction.ex]: Oh right, didn't see the method overloading.

Comment (jeffkreeftmeijer) [inline lib/appsignal/transaction.ex]: We don't really need a variable for `key` in this clause, as we know its value is always `"params"`. I'd prefer explicitly using the string again in the `do_set_sample_data/3` call for clarity.

Comment (tombruijn) [reply lib/appsignal/transaction.ex]: Will change it then 🙂

**Comment by tombruijn** [comment]
Updated the config test to better test if the `APPSIGNAL_SEND_PARAMS` env key works. Will merge if PR is green.

**Pull request merged**